packages/venia-ui/lib/components/ProductOptions/swatchList.js

```diff
@@ -6,25 +6,46 @@ import { useStyle } from '../../classify';
 import defaultClasses from './swatchList.module.css';
 
 const SwatchList = props => {
-    const { getItemKey, selectedValue = {}, items, onSelectionChange } = props;
-
+    const {
+        getItemKey,
+        selectedValue = {},
+        items,
+        onSelectionChange,
+        isALLOutOfStock,
+        isVariantsOutOfStock
+    } = props;
+    let isOptionOutOfStock;
     const classes = useStyle(defaultClasses, props.classes);
-
     const swatches = useMemo(
         () =>
             items.map(item => {
                 const isSelected = item.label === selectedValue.label;
 
+                if (isVariantsOutOfStock && isVariantsOutOfStock.length > 0) {
+                    const flatOosArray = isVariantsOutOfStock.flat();
```

Comment:
I would expand this out to `flatOutOfStockArray` to help reduce the cognitive load for anyone reading through the code

Reply:
Thanks! It's updated.